docs/orchestrating-elastic-stack-applications/agent-fleet.asciidoc

```diff
@@ -0,0 +1,395 @@
+=== Required Kibana configuration
+
+There are two settings that must be set correctly in Kibana configuration for Fleet to work.
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: kibana.k8s.elastic.co/v1
+kind: Kibana
+metadata:
+  name: kibana
+spec:
+  config:
+    xpack.fleet.agents.elasticsearch.host: "https://elasticsearch-es-http.default.svc:9200"
+    xpack.fleet.agents.fleet_server.hosts: ["https://fleet-server-agent-http.default.svc:8220"]
+----
+
+`xpack.fleet.agents.elasticsearch.host` has to point to Elasticsearch cluster that Elastic Agents should send the data to. For Elasticsearch clusters managed by ECK, ECK creates a Service with `https://${ES_RESOURCE_NAME}-es-http.${ES_RESOURCE_NAMESPACE}.svc:9200` url.
+
+`xpack.fleet.agents.fleet_server.hosts` has to point to Fleet Server that Elastic Agents should connect to. For Fleet Server instances managed by ECK, ECK creates a Service with `https://${FS_RESOURCE_NAME}-agent-http.${FS_RESOURCE_NAMESPACE}.svc:8220` url.
+
+[id="{p}-elastic-agent-fleet-configuration-setting-referenced-resources"]
+=== Setting referenced resources
+
+Both Fleet Server and Elastic Agent in Fleet mode can facilitate Fleet setup. Fleet Server can set up Fleet in Kibana (which otherwise requires manual steps) and enroll itself in the default Fleet Server policy. Elastic Agent can enroll itself in the default Elastic Agent policy. To allow ECK to set this up, provide a reference to ECK-managed Kibana through `kibanaRef` configuration element.
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: agent.k8s.elastic.co/v1alpha1
+kind: Agent
+metadata:
+  name: fleet-server
+spec:
+  kibanaRef: kibana
+----
+
+ECK can also facilitate setting up connection between Elastic Agents and ECK-managed Fleet Server. To allow ECK to set this up, provide a reference to Fleet Server through `fleetServerRef` configuration element.
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: agent.k8s.elastic.co/v1alpha1
+kind: Agent
+metadata:
+  name: elastic-agent
+spec:
+  fleetServerRef: fleet-server
+----
+
+
+Set `elasticsearchRefs` element in your Fleet Server to point to the Elasticsearch cluster that will manage Fleet. Leave `elasticsearchRefs` empty or unset for any Elastic Agent running in Fleet mode as the Elasticsearch cluster to target will come from Kibana `xpack.fleet.agents.elasticsearch.host` configuration element.
+
+NOTE: Currently, Elastic Agent in Fleet mode supports only a single output, so only a single Elasticsearch cluster can be referenced.
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: agent.k8s.elastic.co/v1alpha1
+kind: Agent
+metadata:
+  name: fleet-server
+spec:
+  elasticsearchRefs:
+  - name: elasticsearch
+----
+
+By default, every reference targets all nodes in your Elasticsearch, Kibana and Fleet Server deployments, respectively. If you want to direct traffic to specific nodes instead, refer to <<{p}-traffic-splitting>> for more information and examples.
```

Comment:
I think we usually refer to "Kibana instances" in the documentation, "nodes" in the context of Kibana seems odd (at least to me).

```suggestion
By default, every reference targets all instances in your Elasticsearch, Kibana and Fleet Server deployments, respectively. If you want to direct traffic to specific instances instead, refer to <<{p}-traffic-splitting>> for more information and examples.
```